pillowtop/listener.py

```diff
@@ -456,26 +444,21 @@ def __init__(self, create_index=True, online=True, **kwargs):
         super(AliasedElasticPillow, self).__init__(**kwargs)
         # online=False is used in unit tests
         self.online = online
-        index_exists = self.index_exists()
-        if create_index and not index_exists:
-            self.create_index()
-        if self.online and (index_exists or create_index):
-            pillow_logging.info("Pillowtop [%s] Initializing mapping in ES" % self.get_name())
-            self.initialize_mapping_if_necessary()
+        if self.online:
+            index_exists = pillow_index_exists(self)
+            if create_index and not index_exists:
+                create_index_for_pillow(self)
+            if create_index or index_exists:
+                pillow_logging.info("Pillowtop [%s] Initializing mapping in ES" % self.get_name())
+                self.initialize_mapping_if_necessary()
```

Comment:
how come you decided to leave this in the pillow and took the create_index_for_pillow out of it? also, is there ever a time where we'd want to create the index but now the mapping?